### ISSUE TITLE
Rule proposal: `no-useless-interpolation`

### ISSUE THREAD

**Issue by karlhorky** [sindresorhus/eslint-plugin-unicorn#1261]
Hi there! 👋 First of all, thanks again for this plugin, super helpful, the things available in this plugin.

I wanted to propose a new rule to address a pattern that I have seen from a lot of our students at @upleveled - writing overly complex patterns with unnecessary interpolations in template strings (eg. a single string variable, or a string literal, or multiple string literals).

## Fail

```js
const withString = `${str}`;
const withStringLiteral = `${'abc'}`;
const withMultipleStringLiterals = `${'abc'}${'def'}`;
```

## Pass

```js
const withString = str;
const withStringLiteral = 'abc';
const withMultipleStringLiterals = 'abcdef';
```

## Implementation ideas

Looking into the ASTExplorer, I guess the simple version of the first failure case below would be:

- only 1 element in `TemplateLiteral.expressions` (an `Identifier`)
- only 2 elements in `TemplateLiteral.quasis`, all empty strings

For the literals, I guess no limit to the `quasis`, but every element in `expressions` should be a `Literal`.

**Comment by karlhorky** [comment]
Ah and background of a proposal in the ESLint repo: https://github.com/eslint/eslint/issues/10798

**Comment by fisker** [comment]
> Looking into the ASTExplorer, I guess the simple version of the first failure case below would be:

This is not right they are the same  one expression, two empty quasis.

**Comment by fisker** [comment]
> Oh, I just realized this may be difficult to implement without variable type information.

Actually, not really nessary, if it's unknown type, we  can assume it's type-casting, `const foo = String(bar)` still better than template.

**Comment by fisker** [comment]
I think we can accept it.

**Comment by sindresorhus** [comment]
This is now accepted.

**Comment by EvgenyOrekhov** [comment]
Note: the `` `${str}` `` case is already covered by the [`disallowTemplateShorthand`](https://eslint.org/docs/rules/no-implicit-coercion#disallowtemplateshorthand) option of the core `no-implicit-coercion` rule.

**Comment by B2o5T** [comment]
I will work on it!

**Comment by fisker** [comment]
I started few month ago, but I can't remember the progress...

https://github.com/fisker/eslint-plugin-unicorn/commits/no-useless-interpolation

**Comment by B2o5T** [comment]
@fisker Ahh, I sawed your comment too late, already finished 😝

**Comment by damiangreen** [comment]
is anything happening on this?

**Comment by KurtPreston** [comment]
This rule to be implemented by **eslint-plugin-lit**:
https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/no-useless-template-literals.md

**Comment by rentalhost** [comment]
Some progress on that? I could includes functions like `${chalk.bold("Hello")}` to be just `chalk.bold("Hello")`?

**Comment by EvgenyOrekhov** [comment]
Looks like this was implemented in [`@typescript-eslint/no-useless-template-literals`](https://typescript-eslint.io/rules/no-useless-template-literals/)
